### DISCUSSION TITLE
How to upgrade from 1.x to 2.x on a running cluster

### DISCUSSION THREAD

**Discussion by dfoxg** [dswd/vpncloud#292] [Q&A] [answered]
I have 3 vm´s which are running at version 1.4. On top of these vm´s is running a k3s cluster, a mariadb-cluster, glusterfs, .. in short many services which will throw errors if there quorum isn´t met.

Since version 2.x completely changed the network protocol, i´m not able to upgrade one host after another.

@dswd do you have an upgrade-path in mind, without potential destroying my clusters and produce merge/split-brain conflicts and without a complete downtime of my services? 
If not, i must stay on 1.4 which isn´t bad, but not great 👍

**Comment by dswd** [comment, marked as answer]
Hi @dfoxg,

this is a very special situation and migrations without downtime can be quite complicated.
You could setup a 2.x VPN in parallel and switch over gradually by changing the routing or using network bridges depending on whether you are using TUN or TAP.
This is just an idea. I would need to investigate your setup and test-drive this to be sure it works.

**Reply by dfoxg** [reply]
Thanks for your reply. I think I will stay on version 1.4 until a downtime for another reason is necessary. That will reduce the time investment on both sides - but thank you!
